### PR TITLE
Added refactors for the WalletListModal

### PR DESCRIPTION
**Pull request by amanalili**
Supplement to the PR https://github.com/EdgeApp/edge-react-gui/pull/1946

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android

### CODE REVIEW

File: src/components/scenes/FioPendingRequestDetailsScene.js [hunk 1]
```diff
@@ -66,27 +65,19 @@ export class FioPendingRequestDetailsComponent extends Component<Props, LocalSta
...
+    const walletKeys = Object.keys(wallets)
+    const { chain_code, token_code } = selectedFioPendingRequest.content
+    const allowedCurrency = token_code && token_code !== '' ? token_code : chain_code
+    for (const walletKey of walletKeys) {
+      const wallet = wallets[walletKey]
+      const walletCurrencyCodeLowerCase = wallet.currencyCode.toLowerCase()
+      const allowedCurrencyLowerCase = allowedCurrency.toLowerCase()
+      if (walletCurrencyCodeLowerCase === allowedCurrencyLowerCase) {
+        onSelectWallet(wallet.id, wallet.currencyCode)
+        break
```
Comment (amanalili) [inline src/components/scenes/FioPendingRequestDetailsScene.js:78]: Match chain_code to wallet currency code, then check enabled tokens for the token_code if token_code not equal to chain_code and it exist

File: src/components/modals/WalletListModal.js [hunk 2]
```diff
@@ -17,7 +17,16 @@ import { scale } from '../../util/scaling.js'
 import { type TokenSelectObject } from '../common/CryptoExchangeWalletListTokenRow.js'
 import { type AirshipBridge, AirshipModal } from './modalParts.js'
 
-export type WalletListResult = GuiWallet | TokenSelectObject | GuiWalletType | null
+export type WalletListResult = {
+  walletToSelect?: {
+    id: string,
```
Comment (amanalili) [inline src/components/modals/WalletListModal.js:22]: change to walletId

File: src/components/modals/WalletListModal.js [hunk 3]
```diff
@@ -17,7 +17,16 @@ import { scale } from '../../util/scaling.js'
...
+    currencyCode: string
+  },
+  walletToCreate?: {
+    value: string,
```
Comment (amanalili) [inline src/components/modals/WalletListModal.js:26]: change to walletType

File: src/components/common/CryptoExchangeWalletListRow.js [hunk 4]
```diff
@@ -174,8 +174,12 @@ class CryptoExchangeWalletListRow extends Component<Props, LocalState> {
...
-    const checkExcludeCurrencyCodes = excludeCurrencyCodes ? excludeCurrencyCodes.find(currencyCode => currencyCode === wallet.currencyCode) : false
+    const checkAllowedCurrencyCodes = allowedCurrencyCodes
+      ? allowedCurrencyCodes.find(currencyCode => currencyCode.toUpperCase() === wallet.currencyCode.toUpperCase())
+      : true
+    const checkExcludeCurrencyCodes = excludeCurrencyCodes
+      ? excludeCurrencyCodes.find(currencyCode => currencyCode.toUpperCase() === wallet.currencyCode.toUpperCase())
+      : false
     const searchFilterLowerCase = searchFilter.toLowerCase()
     const nameString = wallet.name.toLowerCase()
     const currencyNameString = wallet.currencyNames[wallet.currencyCode].toLowerCase()
```
Comment (amanalili) [inline src/components/common/CryptoExchangeWalletListRow.js:185]: Convert currency arrays on walletListModal to uppercase

File: src/components/modals/WalletListModal.js (hunk 2 above)
Comment (amanalili) [reply src/components/modals/WalletListModal.js]: fixed

File: src/components/modals/WalletListModal.js (hunk 3 above)
Comment (amanalili) [reply src/components/modals/WalletListModal.js]: fixed

File: src/components/common/CryptoExchangeWalletListRow.js (hunk 4 above)
Comment (amanalili) [reply src/components/common/CryptoExchangeWalletListRow.js]: fixed

File: src/components/scenes/FioPendingRequestDetailsScene.js (hunk 1 above)
Comment (amanalili) [reply src/components/scenes/FioPendingRequestDetailsScene.js:86]: fixed

Comment (peachbits) [reply src/components/scenes/FioPendingRequestDetailsScene.js:86]: The first for loop doesn't check if chain_code === wallet.currencyCode. The if(confirmTokenCode) should be moved into the for loop at line 84. The entire first for loop can be removed completely at line 73.

Essentially, only do one for loop that checks if chain_code matches wallet.currencyCode and then checks if confirmTokenCode matches token_code

File: src/components/common/CryptoExchangeWalletListRow.js (hunk 4 above)
Comment (peachbits) [reply src/components/common/CryptoExchangeWalletListRow.js]: This does not appear to be fixed. Currency codes should be preprocessed to uppercase before being used in walletListModal or passed to walletListRow

File: src/components/scenes/FioPendingRequestDetailsScene.js [hunk 5]
```diff
@@ -66,27 +65,27 @@ export class FioPendingRequestDetailsComponent extends Component<Props, LocalSta
...
   setDefaultWallet (): void {
-    const { onSelectWallet } = this.props
-    const { allowedWallets } = this.getWalletsListData()
-    if (allowedWallets && allowedWallets.length) {
-      onSelectWallet(allowedWallets[0].id, allowedWallets[0].currencyCode)
+    const { onSelectWallet, wallets, selectedFioPendingRequest } = this.props
+    const { chain_code, token_code } = selectedFioPendingRequest.content
+    const confirmTokenCode = token_code && token_code !== chain_code ? token_code : null
+    const walletKeys = Object.keys(wallets)
+    if (confirmTokenCode) {
```
Comment (peachbits) [inline src/components/scenes/FioPendingRequestDetailsScene.js:72]: The first for loop doesn't check if chain_code === wallet.currencyCode. The if(confirmTokenCode) should be moved into the for loop at line 84. The entire first for loop can be removed completely at line 73.

Essentially, only do one for loop that checks if chain_code matches wallet.currencyCode and then checks if confirmTokenCode matches token_code

Comment (amanalili) [reply src/components/scenes/FioPendingRequestDetailsScene.js]: updated

**Pull request merged**